figures/mapping-histology-labels.Rmd

```diff
@@ -172,6 +172,37 @@ if (nrow(display_group_df) > 17) {
 }
 ```
 
+# Make `display_order`
+
+Get ranks in order of big to small and make them into a new column in `display_group_df`. 
+We will always want the "normal", "benign", "other_tumor" groups to come last so we are gonna remove them from this ranking and then add back when we get to `histology_table`. 
+
+```{r}
+control_like_groups <- c("normal", "benign", "other tumor")
+
+display_order_df <- display_group_df %>% 
+  dplyr::filter(!display_group %in% control_like_groups) %>%
+  dplyr::mutate(display_order = as.numeric(order(n, decreasing = TRUE))) %>% 
+  dplyr::select(-n)
+```
+
+Add on the `display_order` column using `inner_join`.
+
+```{r}
+max_rank <- max(display_order_df$display_order)
+
+histology_table <- histology_table %>%
+  # Join on the display orders
+  dplyr::left_join(display_order_df, by = "display_group") %>% 
+  # Add ranks for the control like groups
+  dplyr::mutate(display_order = dplyr::case_when(
+    display_group == "benign" ~ max_rank + 1, 
+    display_group == "other tumor" ~ max_rank + 2, 
+    display_group == "normal" ~ max_rank + 3, 
+    TRUE ~ display_order
+    ))
+```
```

Comment:
This ends up a bit convoluted I think. Why do two separate joins rather than sort out  & add the order at the same time as the display_groups earlier?

Also, I think you can simplify a lot and get rid of the confusing `case_when` and `max_rank` stuff:
My suggestion (not checked):
```
display_group_df %>%
  mutate(display_group = forcats::fct_reorder(display_group, n, .desc = TRUE) %>%
                         forcats::fct_relevel("benign", "other tumor", "normal", after = Inf),
         display_order = as.numeric(display_group))
```

Reply:
Ah. This is much better.